include/xrpl/protocol/detail/features.m

```diff
@@ -0,0 +1,112 @@
+//------------------------------------------------------------------------------
```

Comment:
I asked to use a different extension that cannot be confused with a header, because these files are not headers in normal sense of the word (because of macros they rely on) - they cannot be included as-is, they won't parse in an IDE, they cannot be formatted with clang-format. I suggested `.inl` as that's typically used for "inline files" but @thejohnfreeman chose different.

Reply:
It's confusing IMO, but I'm fine with this.

Reply:
This : https://gcc.gnu.org/onlinedocs/cpp/Header-Files.html suggests headers can be used for macros.

Reply:
Yes, headers are often used for macro definitions. Also the language attaches no meaning to file extension. However things get a little tricky for IDEs and other tools when, what appears to be a header file from extension, does not work like one. In this case the apparent header files rely on a macro definitions brought from the outside _without_ the relevant `#include` inside the file, which is not normal practice of how header files work.

If that helps to get the PR approved, I am happy for @thejohnfreeman to rename these files to any extension, including `.h`. I guess it does not matter much.

Reply:
`.m` feels odd but not a showstopper I guess. I'm more concerned about two other issues that I mentioned.

Reply:
Would everyone like `.macro` more than both `.m` and `.h`?

Reply:
If for some reason `.h` is not an option then `.macro` is better. 
But as I mentioned earlier, using macros in general prevents IDE (at least CLion) resolving the fields reference. I find this really inconvenient.